tests/ignite/contrib/handlers/test_lr_finder.py

```diff
@@ -295,6 +296,58 @@ def test_plot(lr_finder, to_save, dummy_engine, dataloader):
     assert not path.exists("/nonexisting/dummy.jpg")
 
 
+def test_plot_multiple_param_groups(lr_finder, dataloader):
+    class Net(nn.Module):
+        def __init__(self):
+            super(Net, self).__init__()
+            self.fc1 = nn.Linear(10, 20)
+            self.fc2 = nn.Linear(20, 10)
+            self.fc3 = nn.Linear(10, 1)
+
+        def forward(self, x):
+            x = F.relu(self.fc1(x))
+            x = F.relu(self.fc2(x))
+            return self.fc3(x)
+
+    model = Net()
+    optimizer = SGD(
+        [
+            {"params": model.fc1.parameters(), "lr": 4e-1},
+            {"params": model.fc2.parameters(), "lr": 3e-2},
+            {"params": model.fc3.parameters(), "lr": 3e-3},
+        ]
+    )
+
+    trainer = create_supervised_trainer(model, optimizer, nn.MSELoss())
+    to_save = {"model": model, "optimizer": optimizer}
+
+    with lr_finder.attach(trainer, to_save) as trainer_with_finder:
+        trainer_with_finder.run(dataloader)
+
+    # Avoid RuntimeError
+    if torch.tensor(lr_finder._history["loss"]).argmin() < 2:
+        lr_finder._history["loss"].insert(0, 10)
+        lr_finder._history["loss"].insert(0, 100)
+        lr_finder._history["lr"].insert(0, [10, 10, 10])
+        lr_finder._history["lr"].insert(0, [100, 100, 100])
```

Comment:
Sometimes we raise this error
```
if len(decreasing_losses) == 1:
            raise RuntimeError("FastaiLRFinder got unexpected curve shape, the curve should be somehow U-shaped")
```
We avoid this RuntimeError by adding some large values at the beginning, and if you're wondering about the shape of these values, in case of multiple `param_groups`, if `len(param_groups) = 3` then `loss = [4, 3, 2, 1]` and `lrs = [[1, 2, 3], [1, 2, 3], [1, 2, 3], [1, 2, 3]]`

Reply:
And regarding the tests, please tell me what do you think about the model and optimizer for `test_plot_multiple_param_groups`, should we move them and make them `pytest.fixture()`? I don't know if we are going to use them in some other tests or not, I think in the next PR `apply_suggested_lr` It will be good to add a test for multiple `param_groups`

Reply:
OK, 
1) the runtime error should be checked in a separate test.
2) let's set up everything such that we do not have runtime error and we do not insert any data a posteriori.

Lines 327-332 looks very strange, no helpful comment explaining why we are doing below lines. This is very bad practice to code like that. Please, avoid that.


> And regarding the tests, please tell me what do you think about the model and optimizer for test_plot_multiple_param_groups, should we move them and make them pytest.fixture()? I don't know if we are going to use them in some other tests or not, I think in the next PR apply_suggested_lr It will be good to add a test for multiple param_groups

Yes, we can factorize them. Good idea !

Reply:
- We already check the runtime error in this test  https://github.com/pytorch/ignite/blob/91953b11a4c44909f6497b6c60c85ee6f3121d41/tests/ignite/contrib/handlers/test_lr_finder.py#L249 . 
- Any idea to reshape the data in that way?
- Fine, I will start working on that refactoring them..

Reply:
> So we better find an idea to save us the inserting part?

If we do not have sufficient lr/loss history, just add more iters to run. Maybe, we can also make the model smaller by reducing the number of hidden layers. I let you explore that.

Reply:
@vfdev-5 Okay I will explore it more, but I don't think if there is any other way to prevent the early points of the loss history to be the minimum, what do you think?

Reply:
I think we can make the NN diverge also by choosing larger max LRs.

Reply:
Okay, I will look into that, Thanks for the ideas!

Reply:
I have been trying to find out a way to avoid this RuntimeError in `test_plot_single_param_group` first, the model already has one layer so we can't reduce it anymore, I tried to manipulate the learning rate with different values, but this didn't work too. I think manipulating the loss history (with the current way or some other way) is the most solid solution

Reply:
> I think manipulating the loss history (with the current way or some other way) is the most solid solution

I would not call it "solid solution". Either you create a synthetic history from the begining or either you work with a model and get real loss values but do not mix both.